src/nvim/main.c

```diff
@@ -225,6 +228,9 @@ void early_init(void)
 
   init_signs();
   ui_comp_syn_init();
+#ifdef WIN32
+  os_dyn_conpty_init();
```

Comment:
Instead of doing this here, `os_dyn_conpty_init` could be called on the first call to `os_has_conpty`. Similar to how `pty_process_spawn` calls `init_termios` its first time.

That avoids adding code and a new import to `main.c`.